addon/components/flash-message.js

```diff
@@ -83,7 +94,11 @@ export default Ember.Component.extend({
   }),
 
   click() {
-    this._destroyFlashMessage();
+    if (this.get('closeFlashOnClick')) {
```

Comment:
Instead of doing conditional logic inside of `click`, you could use the `init` hook:

```js
import Ember from 'ember';

const {
  Component,
  get,
  K
} = Ember;

Component.extend({
  init() {
    this._super(...arguments);
    
    if (get(this, 'flash.closeFlashOnClick')) {
      this.click = K;
    }
  }
});
```